### PR TITLE
Create an output directory in MarkdownReporter

### LINKED ISSUE #394: The command-line tool can create a report directory by itself
When calling the command-line tool to create a report for projects specified in a config, the tool fails if the specified report directory doesn't exist:

```bash
$ java -jar target/fosstars-github-rating-calc.jar  --verbose --token ${TOKEN} --config conf.yml
[+] Loading config from conf.yml
[x] Something went wrong!
java.io.FileNotFoundException: Oh no! I could not find report
	at com.sap.sgs.phosphor.fosstars.tool.github.MarkdownReporter.<init>(MarkdownReporter.java:115)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.makeReporters(SecurityRatingCalculator.java:419)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.processConfig(SecurityRatingCalculator.java:331)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.run(SecurityRatingCalculator.java:236)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.main(SecurityRatingCalculator.java:102)
[x] Bye!
```

The config looks like this:

```
cache: .fosstars/project_rating_cache.json

finder:
  repositories:
    - organization: apache
      name: commons-io
    - organization: apache
      name: httpcomponents-client

reports:
  - type: json
    where: github_projects.json
  - type: markdown
    source: github_projects.json
    where: report

```

The command-line tool should create the `report` directory by itself.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of updates:

- Updated `MarkdownReporter` to create an output directory if it doesn't exist.
- Added a test.

Fixes #394

### CODE REVIEW

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/github/MarkdownReporter.java [hunk 1]
```diff
@@ -130,14 +129,22 @@
...
+
+    Path directory = Paths.get(outputDirectory);
+    if (Files.isSymbolicLink(directory) || Files.isRegularFile(directory)) {
+      throw new IOException(
+          String.format("Oops! Output directory is not a directory: %s", outputDirectory));
+    }
+    if (!Files.isDirectory(directory)) {
+      Files.createDirectories(directory);
+    }
+    if (!Files.isDirectory(directory)) {
```
Comment (Sachpat) [inline src/main/java/com/sap/oss/phosphor/fosstars/tool/github/MarkdownReporter.java:142]: Isn't the condition on line https://github.com/SAP/fosstars-rating-core/pull/415/files#diff-0d1a21dc11fa7ecb0693a45578821f78d8b39558d2c0cdac1cdfaea7065aadf9R139 same as this?

Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/github/MarkdownReporter.java:142]: If the path is a symlink of a file, it doesn't make sense to continue. Then, the directory may or may not exist. Here we check it.

Comment (Sachpat) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/github/MarkdownReporter.java:142]: No , I mean, just above this condition, we check the same and create right? then why again checking it?

Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/github/MarkdownReporter.java:142]: It makes sure that the directory was created. This check it a bit paranoid and therefore may look redundant.

**Review by Sachpat — APPROVED** [review]
Looks good to me.

**Comment by artem-smotrakov** [comment]
Thanks for the review @Sachpat !

**Pull request merged**